lib/publish.js

```diff
@@ -30,53 +30,78 @@ module.exports = async (pluginConfig, context) => {
   debug('milestones: %o', milestones);
 
   if (assets && assets.length > 0) {
-    const globbedAssets = await getAssets(context, assets);
+    // Skip glob if url is provided
+    const urlAssets = assets.filter(asset => asset.url);
+    debug('url assets: %o', urlAssets);
+    const globbedAssets = await getAssets(
+      context,
+      assets.filter(asset => !asset.url)
+    );
     debug('globbed assets: %o', globbedAssets);
+    const allAssets = [...urlAssets, ...globbedAssets];
+    debug('all assets: %o', allAssets);
 
     await Promise.all(
-      globbedAssets.map(async asset => {
+      allAssets.map(async asset => {
         const {path, label, type, filepath} = isPlainObject(asset) ? asset : {path: asset};
-        const file = resolve(cwd, path);
-        let fileStat;
-
-        try {
-          fileStat = await stat(file);
-        } catch (_) {
-          logger.error('The asset %s cannot be read, and will be ignored.', path);
-          return;
-        }
-
-        if (!fileStat || !fileStat.isFile()) {
-          logger.error('The asset %s is not a file, and will be ignored.', path);
-          return;
-        }
-
-        debug('file path: %o', path);
-        debug('file label: %o', label);
-        debug('file type: %o', type);
-        debug('file filepath: %o', filepath);
-
-        // Uploaded assets to the project
-        const form = new FormData();
-        form.append('file', createReadStream(file));
-
-        const uploadEndpoint = urlJoin(gitlabApiUrl, `/projects/${encodedRepoId}/uploads`);
-
-        debug('POST-ing the file %s to %s', file, uploadEndpoint);
-
-        let response;
-        try {
-          response = await got.post(uploadEndpoint, {...apiOptions, body: form}).json();
-        } catch (error) {
-          logger.error('An error occurred while uploading %s to the GitLab project uploads API:\n%O', file, error);
-          throw error;
+        let _url = asset.url;
+        if (_url) {
+          // Allow loading variable from process.env load CI_JOB_ID for attaching CI artifact
```

Comment:
Nice you considered this, too 👍

Reply:
I understand the idea behind this, but I don't like how this new propery is handled in a very special way. We already have a templating mechanism (via lodash) and I don't want to introduce another one. I'd suggest we switch to that for the URL and create a follow-up issue for using env vars in templates.